### PR TITLE
feat(analytics): Add tracking for source map errors

### PR DESCRIPTION
**Pull request by mikellykels**
Add tracking for what errors people are running into when using source maps.

cc @ceorourke

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/eventEntries.jsx [hunk 1]
```diff
@@ -67,6 +68,25 @@ const EventEntries = createReactClass({
     };
   },
 
+  recordSourcemapError(errorType) {
+    analytics('sourcemap.sourcemap_error', {
```
Comment (denamwangi) [inline src/sentry/static/sentry/app/components/events/eventEntries.jsx]: nit: one cool thing you could do here is take advantage of es6 destructuring and do 
` let {project, event} = this.props;` which will let you pull just `project.organization.id` and `event.groupID`

File: src/sentry/static/sentry/app/components/events/eventEntries.jsx [hunk 2]
```diff
@@ -67,6 +68,25 @@ const EventEntries = createReactClass({
...
+      'org_id': this.props.project.organization.id,
+      'group': this.props.event.groupID,
+      'error_type': errorType
+    })
+  },
+
+  componentDidMount() {
+    if (typeof this.props.event.errors !== 'undefined') {
```
Comment (denamwangi) [inline src/sentry/static/sentry/app/components/events/eventEntries.jsx]: same deal here with destructuring `let {event} = this.props;`

File: src/sentry/static/sentry/app/components/events/eventEntries.jsx [hunk 3]
```diff
@@ -67,6 +68,25 @@ const EventEntries = createReactClass({
...
+      let errorTypes = this.props.event.errors;
+      let errorType = [];
+      for (let i = 0; i < errorTypes.length; i++) {
+        errorType.push(errorTypes[i].type);
```
Comment (denamwangi) [inline src/sentry/static/sentry/app/components/events/eventEntries.jsx]: a cool thing you can do here is take advantage of `.map()` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map to populate your array for you

something like `let x = errorTypes.map((errorType) => errorType.type)`

File: src/sentry/static/sentry/app/components/events/eventEntries.jsx (hunk 2 above)
Comment (denamwangi) [reply src/sentry/static/sentry/app/components/events/eventEntries.jsx]: For this check a pattern we use a bunch is: 
`if (!this.props.event.errors) return;`

**Review by denamwangi — APPROVED** [review]
These aren't blockers by any means but i'd suggest a couple of cosmetic changes

**Comment by mikellykels** [comment]
linking to jira ticket https://getsentry.atlassian.net/browse/SE-53

**Pull request merged**